src/WebJobs.Extensions.DurableTask/Listener/TaskActorShim.cs

```diff
@@ -61,22 +60,24 @@ public override RegisteredFunctionInfo GetFunctionInfo()
 
         public override string GetStatus()
         {
-            if (this.lastStartedOperation.IsCompleted)
+            this.context.CurrentStateView?.WriteBack();
+
+            if (!this.context.State.ActorExists || this.context.State.ActorState == null)
             {
-                // the actor is not currently processing an operation
-                return "Available";
+                return "";
             }
             else
             {
-                if (this.context.State.Queue != null)
+                int payloadSizeInKB = (int)(Encoding.Unicode.GetByteCount(this.context.State.ActorState) / 1024.0);
+
+                if (payloadSizeInKB <= 16)
                 {
-                    // the actor is currently processing an operation, and more operations are waiting in the queue)
-                    return $"Busy ({this.context.State.Queue.Count} queued)";
+                    return this.context.State.ActorState;
                 }
                 else
                 {
-                    // the actor is currently processing an operation
-                    return "Busy";
+                    // this is not a valid JSON string so it cannot be mistaken for the actual actor state
+                    return $"Large ({payloadSizeInKB} KB)";
```

Comment:
Yes, this is a current limitation. I am looking into removing it. There are a lot of overloads for getting the status of an orchestration, perhaps I can find one that the client can call and that retrieves the input column, then I can use that directly instead of going via the status column.

Reply:
Looks like I can fix that pretty easily, the input is already being returned by this query.

Reply:
Perhaps we should extend this interface a bit so it returns more status information about the actor, not just its state? For example, we could also return (1) whether the actor exists, (2) whether the actor is locked by an orchestration, and (3) the size of the queue.

Reply:
I actually think the bits of information (2) and (3) are not a good idea because they are only eventually consistent. But (1) is necessary I think so that users can distinguish between an actor that does not exist, and an actor that is in default state.